hyperspy/signal.py

```diff
@@ -4111,45 +4114,71 @@ def add_marker(
         >>> marker.name = "point_marker"
         >>> s.add_marker(marker, permanent=True)
         >>> del s.metadata.Markers.point_marker
+
+        Adding many markers as a list
+        >>> from numpy.random import random
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(100, 100)))
+        >>> marker_list = []
+        >>> for i in range(100):
+        >>>     marker = hs.markers.point(random()*100, random()*100, color='red')
+        >>>     marker_list.append(marker)
+        >>> s.add_marker(marker_list, permanent=True)
+
         """
-        marker_data_shape = marker._get_data_shape()
-        if (not (len(marker_data_shape) == 0)) and (
-                marker_data_shape != self.axes_manager.navigation_shape):
-            raise ValueError(
-                "Navigation shape of the marker must be 0 or the "
-                "same navigation shape as this signal.")
-        if (marker.signal is not None) and (marker.signal is not self):
-            raise ValueError("Markers can not be added to several signals")
-        marker._plot_on_signal = plot_on_signal
-        if plot_marker:
-            if self._plot is None:
-                self.plot()
-            if marker._plot_on_signal:
-                self._plot.signal_plot.add_marker(marker)
-            else:
-                if self._plot.navigator_plot is None:
-                    self.plot()
-                self._plot.navigator_plot.add_marker(marker)
-            marker.plot()
+        if issubclass(marker.__class__, MarkerBase):
```

Comment:
Fixed! But I'm using the `issubclass(marker.__class__, MarkerBase)` when plotting the the permanent markers, since other stuff can (I guess) fairly easily sneak into the metadata.Markers.